lib/services/role.go

```diff
@@ -2515,6 +2559,11 @@ type AccessMFAParams struct {
 	// AlwaysRequired is set when MFA is required for all sessions, regardless
 	// of per-role options.
 	AlwaysRequired bool
+	// NeverRequired is set when MFA is never required for any sessions, regardless
+	// of per-role options. This means either both the cluster auth preference and
+	// all roles have per-session MFA off, or at least one of those resources has
+	// "require_session_mfa: hardware_key_touch", which overrides per-session MFA.
+	NeverRequired bool
```

Comment:
I think it's a bit confusing to have `AlwaysRequired` and `NeverRequired` options here. Can we create maybe a new type/enum with values like `Unknown`, `Always`, `Never` and use only one field, `Required`? What if someone sets both of them to `true`, for example?